sarif/sarif_file_utils.py

```diff
@@ -100,6 +100,103 @@ def read_result_location(result) -> Tuple[str, str]:
     return (file_path, line_number)
 
 
+def read_result_rule(result, run) -> Tuple[Union[dict, None], int]:
+    """
+    Return's the corresponding rule object for the specified result, plus its index
+    in the rules array. Follows the rules at
+    https://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html#_Toc141790895
+    """
+    ruleIndex = result.get("ruleIndex", None)
```

Comment:
`None` is the default fallback value for `get`.  I think it's nice to include it explicitly (others disagree), but there are other places where you've just used `get(key)` - it would be good to be consistent.  (The existing code probably isn't!!)